### PR TITLE
Introduce RandomMemInstruction generator functions and make Load operations only take src registers

### PR DESCRIPTION
**Pull request by thatjiaozi**
Introduce helper functions for generating random memory storing/loading operations. Also delete useless code around load operations: under normal circumstances load operations can only take registers as src values.

### CODE REVIEW

File: pkg/ebpf/instruction_generators.go [hunk 1]
```diff
@@ -76,6 +76,90 @@ func RandomJmpInstruction(maxOffset uint64) *pb.Instruction {
...
+func RandomOffset(s pb.StLdSize) int16 {
+	// Cap offsets to 512.
+	maxOffset := int16(512)
+	offset := int16(rand.SharedRNG.RandInt()) % maxOffset
+	for offset == 0 {
+		offset = int16(rand.SharedRNG.RandInt()) % maxOffset
+	}
+
+	if offset > 0 {
+		// Mem offsets from the stackc an only be negative.
```
Comment (amlweems) [inline pkg/ebpf/instruction_generators.go:119]: `stackc an` -> `stack can`

**Review by amlweems — APPROVED** [review]
lgtm!

File: pkg/ebpf/instruction_generators.go (hunk 1 above)
Comment (thatjiaozi) [reply pkg/ebpf/instruction_generators.go]: woops, done.

**Comment by thatjiaozi** [comment]
thanks!

**Pull request merged**